runtime/common/EvolveResult.h

```diff
@@ -20,61 +20,62 @@ namespace cudaq {
 /// and any computed expectation values during and after evolution,
 /// depending on the arguments passed to the call.
 class evolve_result {
-protected:
-  // The final state after the time evolution.
-  state final_state;
-
-  // The state after each step in the evolution, if computed.
-  std::optional<std::vector<state>> intermediate_states = {};
-
-  // The computed expectation values at the end of the evolution.
-  std::optional<std::vector<observe_result>> final_expectation_values = {};
+public:
+  // The state of the system. If only final state is retained, this vector will
+  // have exactly one element.
+  std::optional<std::vector<state>> states = std::nullopt;
 
-  // The computed expectation values for each step of the evolution.
+  // The computed expectation values. If only final expectation values are
+  // retained, this vector will have exactly one element.
   std::optional<std::vector<std::vector<observe_result>>> expectation_values =
-      {};
+      std::nullopt;
 
   // The result of sampling of an analog Hamiltonian simulation on a QPU
-  std::optional<sample_result> sampling_result = {};
+  std::optional<sample_result> sampling_result = std::nullopt;
 
-public:
-  evolve_result(state state) : final_state(state) {}
+  // Construct from single final state.
+  evolve_result(state state) {}
 
-  evolve_result(state state, const std::vector<observe_result> &expectations)
-      : final_state(state),
-        final_expectation_values(
-            std::make_optional<std::vector<observe_result>>(expectations)) {}
+  // Construct from single final observe result.
+  evolve_result(state state, const std::vector<observe_result> &expectations) {
```

Comment:
Did we land on having the return types for expectation be std::optional then? 
Works for me, just double checking.